nipy/algorithms/clustering/imm.py

```diff
@@ -234,7 +236,7 @@ def cross_validated_update(self, x, z, plike, kfold=10):
         if np.isscalar(kfold):
             aux = np.argsort(np.random.rand(n_samples))
             idx = - np.ones(n_samples).astype(np.int)
-            j = np.ceil(n_samples / kfold)
+            j = int(math.ceil(n_samples / kfold))
```

Comment:
If `kfold` doesn't divide evenly, it will be off by one:

Suppose `n_samples = 5`, `kfold = 2`.

Python 2: `int(math.ceil(5 / 2)) == int(math.ceil(2)) == int(2.0) == 2`
Python 3: `int(math.ceil(5 / 2)) == int(math.ceil(2.5)) = int(3.0) == 3`